exporter/otlpexporter/internal/arrow/stream.go

```diff
@@ -215,12 +225,31 @@ func (s *Stream) write(ctx context.Context) {
 		// Let the receiver knows what to look for.
 		s.setBatchChannel(batch.BatchId, wri.errCh)
 
+		// Optionally include outgoing context, if present.  Note that if
```

Comment:
This explanation was really helpful.

Reply:
Thanks, but I've _totally changed this logic_ thanks to Gustavo's insight.